src/main/scala/org/constellation/domain/healthcheck/ReconciliationRound.scala

```diff
@@ -0,0 +1,244 @@
+
+  val retryAttempts = 2
+  val retryAfter: FiniteDuration = 60.seconds
+
+  private[healthcheck] val reconciliationResult: Ref[F, Option[ReconciliationResult]] =
+    Ref.unsafe(None)
+
+//  private[healthcheck] val firstClusterAlignmentResults: Ref[F, Option[Map[Id, List[ClusterAlignmentResult]]]] =
+//    Ref.unsafe(None)
+
+//  private[healthcheck] val lastClusterAlignmentResults: Ref[F, Option[Map[Id, List[ClusterAlignmentResult]]]] =
+//    Ref.unsafe(None)
+
+  def getReconciliationResult(): F[Option[ReconciliationResult]] = reconciliationResult.get
+
+  def start(): F[Fiber[F, Unit]] = F.start(run())
+
+  private def run(
+    attemptsLeft: Int = retryAttempts,
+    firstClusterAlignmentResult: Option[Map[Id, List[ClusterAlignmentResult]]] = None,
+    lastClusterAlignmentResult: Option[Map[Id, List[ClusterAlignmentResult]]] = None
+  ): F[Unit] =
+    for {
+      isFirstRun <- F.delay(attemptsLeft == retryAttempts)
+      isLastRun <- F.delay(attemptsLeft == 0)
+      _ <- if (isFirstRun) F.unit else T.sleep(retryAfter)
+      result <- runReconciliation()
+      _ <- result match {
+        case Left(unresponsivePeers) =>
+          reconciliationResult.modify(
+            // shouldn't we only run healthcheck for nodes that didn't respond instead of running consensus right away?
+            // in consequence I think we should also try to calculate cluster state from the responses we got and start consensuses for these
+            _ => (ReconciliationResult(unresponsivePeers, Set.empty, lastClusterAlignmentResult).some, ())
+          )
+
+        case Right(clusterAlignmentResults) =>
+          if (areAllNodesAligned(clusterAlignmentResults))
+            reconciliationResult.modify(
+              _ => (ReconciliationResult(Set.empty, Set.empty, clusterAlignmentResults.some).some, ())
+            )
+          else if (!isLastRun)
+            run(
+              attemptsLeft - 1,
+              if (isFirstRun) clusterAlignmentResults.some else firstClusterAlignmentResult,
+              clusterAlignmentResults.some
+            )
+          else {
+            for {
+              initiallyMisaligned <- F.delay(
+                firstClusterAlignmentResult.map(fetchMisalignedPeers).getOrElse(Set.empty)
+              )
+              currentlyMisaligned <- F.delay(fetchMisalignedPeers(clusterAlignmentResults))
+              misaligned = initiallyMisaligned.intersect(currentlyMisaligned)
+              _ <- reconciliationResult.modify(
+                _ => (ReconciliationResult(Set.empty, misaligned, clusterAlignmentResults.some).some, ())
+              )
+            } yield ()
+          }
+      }
+    } yield ()
+
+  def areAllNodesAligned(clusterAlignmentResults: Map[Id, List[ClusterAlignmentResult]]): Boolean =
+    clusterAlignmentResults.forall { case (_, result) => result == List(NodeAligned) }
+
+  def fetchMisalignedPeers(clusterAlignmentResults: Map[Id, List[ClusterAlignmentResult]]): Set[Id] =
```

Comment:
It does not fetch anything and is pure ;p

Reply:
I will reconsider the naming :).